lib/storage.js

```diff
@@ -1,18 +1,32 @@
-const raf = require('random-access-file')
-const Corestore = require('corestore')
-const join = require('path').join
+const raf = require('random-access-file');
+const Corestore = require('corestore');
+const join = require('path').join;
+const fsctl = require('fsctl');
 
-module.exports = function defaultCorestore (storage, opts) {
-  if (isCorestore(storage)) return storage
-  
+module.exports = function defaultCorestore(storage, opts) {
+  if (isCorestore(storage)) return storage;
+  let factory = null;
+  console.log(storage);
   switch (typeof storage) {
-    case 'function': return new Corestore(path => storage(path), opts)
-    case 'string': return new Corestore(path => raf(join(storage, path), opts))
+    case 'function':
+      return new Corestore((path) => storage(path), opts);
+    case 'string':
+      const sparseWin =
+        opts.sparseWin && process.platform === 'win32'
+          ? fsctl.sparse
+          : opts.sparse;
+      return new Corestore(
+        (path) => raf(join(storage, path), { ...opts, sparse: sparseWin }),
+        opts
+      );
```

Comment:
remove all semicolons in general

Reply:
And maybe add a comment saying this is a temp workaround for windows sparse files until it lands as the default here.